kong/init.lua

```diff
@@ -115,6 +115,31 @@ local declarative_entities
 local schema_state
 
 
+local reset_kong_shm
+do
+  local preserve_keys = {
+    "events:requests",
+    "kong:node_id",
```

Comment:
Need to discuss how to handle declarative config here.

Reply:
Nothing needs to be done. :tada: If we make it remove the `declarative_config:loaded` key, then `Kong.init()` will naturally reload the declarative config YAML file from `declarative_config` in `kong.conf`, just as we wanted.

Reply:
FWIW, I just tested this locally in `database=off` with and without a `declarative_config` entry, by adding `ngx.shared.kong:delete("declarative_config:loaded")` at the beginning of `Kong.init()`.

With a `declarative_config` entry:

1. `KONG_DATABASE=off KONG_DECLARATIVE_CONFIG=kong.yml KONG_PREFIX=foo kong start`
2. `sed -i 's/my-service/mon-service/g' kong.yml`
3. `kong reload -p foo`
4. `http :8001/services | grep mon-service`

Without a `declarative_config` entry, it still preserves whatever is in memory, which is nice:

1. `KONG_DATABASE=off KONG_PREFIX=foo kong start`
2. `http :8001/config config=@kong.yml`
3. `kong reload -p foo`
4. `http :8001/services | grep mon-service`

(Now that I think of it, these would be nice test cases to have)